### PR TITLE
chore: Readability and comprehensibility of code

### PR DESCRIPTION
**Pull request by Tessg22**
- Add dry_run argument
- Rename variables
- Put if statement on line 120 into function for better comprehensibility of code

### CODE REVIEW

File: scripts/lightweight_curator.py [hunk 1]
```diff
@@ -84,7 +85,7 @@ def get_max_allowed_size(es, percentage_threshold):
     for node in data.splitlines():
         i = i + int(node)
 
-    max_allowed_size = (percentage_threshold * i) / 100.0
+    max_allowed_size = int( (percentage_threshold * i) / 100.0 )
```
Comment (georgettica) [inline scripts/lightweight_curator.py:88]: Why should this be int?
I don't remember the functionality but in some languages converting 78.5 to int returns 78
If this is intentional ingore

File: scripts/lightweight_curator.py [hunk 2]
```diff
@@ -98,33 +99,40 @@ def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
     """
     Returns a list of actionable indices based on percentage size and age.
     """
-    class data_structure:
+    class indices_struct:
```
Comment (georgettica) [inline scripts/lightweight_curator.py]: Would rename to index but not critical

File: scripts/lightweight_curator.py [hunk 3]
```diff
@@ -98,33 +99,40 @@ def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
...
+    Returns list of indices which are above threshold limit.
+    """
     indices_to_delete = []
-    for data_object in sorted(data_array, key=lambda x: x.creation_date, reverse=True):
-        if size_counter < max_allowed_size and data_object.size + size_counter < max_allowed_size:
-            log(log_info, "Do not add into actionable list: '{indice}', summed disk usage is {usage} B and disk limit is {limit} B".format(
-                indice=data_object.name, usage=size_counter, limit=int(max_allowed_size)))
-            size_counter += data_object.size
+    indices_size_counter = 0
+    for index in sorted(indices, key=lambda x: x.creation_date, reverse=True):
```
Comment (georgettica) [inline scripts/lightweight_curator.py:139]: Should maybe add a comment on the reason for reverse true

File: scripts/lightweight_curator.py [hunk 4]
```diff
@@ -98,33 +99,40 @@ def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
...
+        expected_size = index.size + indices_size_counter
+        def index_smaller_then_max_allowed_size(indices_size_counter, expected_size, max_allowed_size):
+            if indices_size_counter < max_allowed_size and expected_size < max_allowed_size:
```
Comment (georgettica) [inline scripts/lightweight_curator.py]: I would just return the conditions in the if, it makes the code smaller and handles the false case aswell

Comment (georgettica) [reply scripts/lightweight_curator.py]: Also maybe move the function somewhere else

File: scripts/lightweight_curator.py [hunk 5]
```diff
@@ -98,33 +99,40 @@ def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
...
+                return True
+        if index_smaller_then_max_allowed_size(indices_size_counter, expected_size, max_allowed_size):
+            log(log_info, "Do not add into actionable list: '{index}', summed disk usage is {usage} B and disk limit is {limit} B".format(
```
Comment (georgettica) [inline scripts/lightweight_curator.py]: Might I suggest using f strings if you are using an updated python version

https://www.geeksforgeeks.org/formatted-string-literals-f-strings-python/

**Review by georgettica — COMMENTED** [review]
Have some stuff I found here, home my input helps

File: scripts/lightweight_curator.py [hunk 6]
```diff
@@ -158,7 +167,7 @@ def main():
     # Pass elasticsearch connect arguments.
     es = es_connect_args(elasticsearch_host)
 
-    log(log_info, "Removing indices which are above {percentage} threshold from host '{host}'".format(
+    log(log_info, "Searching for indices which are above {percentage}% threshold from host '{host}'".format(
```
Comment (drewandersonnz) [inline scripts/lightweight_curator.py]: This states you are searching for indices which themselves are individually above threshold. 

Maybe say something like "Removing/Searching for oldest indices to remove first when total storage exceeds threshold".

File: scripts/lightweight_curator.py (hunk 1 above)
Comment (Tessg22) [reply scripts/lightweight_curator.py:88]: On bellow line I had previously _limit=int(max_allowed_size)_ for the reason that units are already in bytes. And I am deleting based on bytes. Imho there is no reason for displaying max_allowed_size as float. Just round the max_allowed_size.

`log(log_info, "Do not add into actionable list: '{index}', summed disk usage is {usage} B and disk limit is {limit} B".format(
                index=index.name, usage=indices_size_counter, limit=max_allowed_size))`

I would still like to know your opinion on this Ron.

File: scripts/lightweight_curator.py (hunk 2 above)
Comment (Tessg22) [reply scripts/lightweight_curator.py]: Done

File: scripts/lightweight_curator.py (hunk 3 above)
Comment (Tessg22) [reply scripts/lightweight_curator.py:139]: Done.

File: scripts/lightweight_curator.py (hunk 5 above)
Comment (Tessg22) [reply scripts/lightweight_curator.py]: Done.

File: scripts/lightweight_curator.py (hunk 4 above)
Comment (Tessg22) [reply scripts/lightweight_curator.py]: Could you please review code if its the way you suggested now?

File: scripts/lightweight_curator.py (hunk 6 above)
Comment (Tessg22) [reply scripts/lightweight_curator.py]: I took inspiration from your suggestion.
Could you please check if it looks good now?

**Review by drewandersonnz — APPROVED** [review]
my request has been done

File: scripts/lightweight_curator.py (hunk 1 above)
Comment (georgettica) [reply scripts/lightweight_curator.py:88]: I stand corrected, and thanks for the elaborate response.
This should be in bytes and shouldn't be a float

File: scripts/lightweight_curator.py [hunk 7]
```diff
@@ -94,37 +95,49 @@ def get_first_item(a_dict={}):
     first_value = next(value_iterator)
     return first_value
 
+def index_smaller_then_max_allowed_size(index, limit, indices_size_counter, indices_to_delete):
```
Comment (georgettica) [inline scripts/lightweight_curator.py]: If this returns a list then imo it shou be plural
I know it's very confusing where to put which term but you get the hang of it after a while
```suggestion
def indices_smaller_then_max_allowed_size(index, limit, indices_size_counter, indices_to_delete):
```

File: scripts/lightweight_curator.py [hunk 8]
```diff
@@ -94,37 +95,49 @@ def get_first_item(a_dict={}):
...
+    if indices_size_counter < limit and expected_size < limit:
+        indices_size_counter += index.size
+        log(log_info, f"Do not add into actionable list: {index.name}, summed disk usage is {indices_size_counter} B and disk limit is {limit} B")
+    else:
+        log(log_info, f"Add into actionable list: {index.name}, summed disk usage is {indices_size_counter} B and disk limit is {limit} B")
+        indices_to_delete.append(index.name)
+
+    return indices_to_delete
+
 def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
```
Comment (georgettica) [inline scripts/lightweight_curator.py]:
```suggestion
def get_actionable_indices(es, max_allowed_size, index_name_prefixes):
```

File: scripts/lightweight_curator.py [hunk 9]
```diff
@@ -94,37 +95,49 @@ def get_first_item(a_dict={}):
...
 
-    # Prepare dictionary of indices with their size and creation_date values.
-    size_counter = 0
-    data_array = []
-
+    """
+    Appends index into the list of indices with their name, size and creation_date.
+    """
+    indices = []
     for index_name_prefixes in index_name_prefixes_list:
```
Comment (georgettica) [inline scripts/lightweight_curator.py]:
```suggestion
    for index_name_prefix in index_name_prefixes:
```

**Review by georgettica — COMMENTED** [review]
Very tiny nits.

No need for second pass from my side after you fix them.

Can be approved and fixed later if you want this PR merged sooner rather than later

**Pull request merged**